src/utils/dockerfileUtils.ts

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Dockerfile, DockerfileParser, Keyword } from 'dockerfile-ast';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { localize } from '../localize';
+
+export interface DockerfileInfo {
+    rootFolder?: string;
+    dockerfileNameRelativeToRoot: string;
+    ports?: number[];
+}
+
+export async function parseDockerfile(rootFolderPath: string, dockerfilePath: string): Promise<DockerfileInfo> {
+    const content = await readFileContent(dockerfilePath);
+    const dockerFile = DockerfileParser.parse(content);
+    const ports: number[] = await getExposedPorts(dockerFile);
+    let dockerFilenameRelativeToRoot = path.relative(rootFolderPath, dockerfilePath);
+    // Ensure the path uses POSIX separators (which work on both Linux and Windows).
+    dockerFilenameRelativeToRoot = dockerFilenameRelativeToRoot.replace(/\\/g, '/');
```

Comment:
I think `pathNormalize(dockerFilenameRelativeToRoot, 'Linux')` can be used instead of the regex replace.